ios/RNSScreen.mm

```diff
@@ -1198,6 +1198,29 @@ @implementation RNSScreenManager
 RCT_EXPORT_VIEW_PROPERTY(homeIndicatorHidden, BOOL)
 #endif
 
+#if !TARGET_OS_TV
+// See:
+// 1. https://github.com/software-mansion/react-native-screens/pull/1543
+// 2. https://github.com/software-mansion/react-native-screens/pull/1596
+// This class is instatiated from React Native's internals during application startup
+- (instancetype)init
+{
+  if (self = [super init]) {
+    dispatch_async(dispatch_get_main_queue(), ^{
+      [[UIDevice currentDevice] beginGeneratingDeviceOrientationNotifications];
+    });
+  }
+  return self;
+}
+
+- (void)dealloc
+{
+  dispatch_sync(dispatch_get_main_queue(), ^{
+    [[UIDevice currentDevice] endGeneratingDeviceOrientationNotifications];
+  });
+}
+#endif // !TARGET_OS_TV
+
```

Comment:
I decided to add this code to `RNSScreenManager`. 
I'm asking for review, to make sure that there aren't any quirks in view manager's lifecycle.

Reply:
Tbh, I am not familiar with the lifecycle of view managers, did you test how does it work on Fabric?

Reply:
Yep, I added ad breakpoint at `init` method and it got called during initial app loading.